src/wallet/coinselection.cpp

```diff
@@ -171,6 +168,24 @@ bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& target_v
     return true;
 }
 
+bool SelectCoinsSRD(std::vector<OutputGroup>& utxo_pool, const CAmount& target_value, std::set<CInputCoin>& out_set, CAmount& value_ret)
```

Comment:
Consider returning a `std::optional<std::pair<std::set<CInputCoin>, CAmount>>` rather than using out references and returning a bool to indicate success.

Reply:
Done